setup.py

```diff
@@ -41,6 +41,7 @@ def read(fname):
     'moto[server]',
     'pathlib2',
     'responses',
+    'boto',
```

Comment:
Why do our tests still need boto?

Reply:
There's a `test_old` method which uses an old `boto` bucket: https://github.com/RaRe-Technologies/smart_open/blob/ecce29ebb3b0dee519f6939e924ee4bb5974ca33/smart_open/tests/test_s3.py#L576

I could at least leave a comment here with an explanation. I suppose alternatively the support for passing in old boto buckets could be deprecated and the test removed.

Reply:
Yeah, I think it's best to remove it altogether.

Sorry for the late response.